src/Model/Export/Catalog/ProductField/GenericField.php

```diff
@@ -53,7 +54,13 @@ public function getName(): string
      */
     public function getValue(AbstractModel $product): string
     {
-        return implode('|', $this->valuesExtractor->getAttributeValues($product, $this->getAttribute()));
+        $value = implode('|', $this->valuesExtractor->getAttributeValues($product, $this->getAttribute()));
+
+        if ($this->getAttribute()->getBackendModel() == 'Magento\Eav\Model\Entity\Attribute\Backend\Datetime' && !empty($value)) {
```

Comment:
I think this if statement should be moved to the AttributeValuesExtractor. Is `getFrontendInput` of attribute of type DateTime is the same as the regular text attribute?

Reply:
if statement moved to AttributeValuesExtractor. Generally there are 2 date type attributes in magento - Date and Date and Time which has getFrontendInput values `date` and `datetime` respectively  so I have add 2 separate cases for each value